### DISCUSSION TITLE
How to change lsp server root_dir?

### DISCUSSION THREAD

**Discussion by jhartum** [LazyVim/LazyVim#181] [Q&A]
I try to change root_dir in LSP Configuration by this way but root_dir does not change

**.config/nvim/lua/plugins/config.lua**
```
return {
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {


      ---@type lspconfig.options

      servers = {
        -- pyright will be automatically installed with mason and loaded with lspconfig
        sourcery = {
          root_dir = function()
            return "/Users/jhartum/Desktop/project/backend/app"
          end,
        },
        pylsp = {
          root_dir = function()
            return "/Users/jhartum/Desktop/project/backend/app"
          end,
          settings = {
            pylsp = {
              plugins = {
                configurationSources = { "mypy", "ruff", "black", "isort", "rope_autoimport" },
                isort = { enabled = true },
                black = { enabled = true },
                ruff = { enabled = true },
                mypy = { enabled = true },
                rope_autoimport = { enabled = true },
                flake8 = { enabled = false, maxLineLength = 50 },

                yapf = { enabled = false },
                autopep8 = { enabled = false },
                pycodestyle = { enabled = false },
                pyflakes = { enabled = false },
              },
            },
          },
        },
      },
    },
  },
}
```

**Comment by folke** [comment]
You probably have  a `.neoconf.json` file in that directory? If so, then that would be the root dir.

**Reply by jhartum** [reply]
Nope, I don`t have a .neoconf.json file

**Reply by folke** [reply]
Well, you must have something that changes the root_dir. I just tested this myself, and it works as expected. Maybe you have a `.vscode` directory?

**Reply by jhartum** [reply]
yes, I have that file(my team use VSCode)
is it possible to change root_dir without moving .vscode file?

**Reply by folke** [reply]
LazyVim includes https://github.com/folke/neoconf.nvim

You can disable importing of vscode stuff by setting `opts.import.vscode = false` for neoconf

**Reply by jhartum** [reply]
I use python-lsp-rope for autoimports, but LSP server root_dir wrong and import wrong too
always I set root_dir in lspconfig, but in lazy vim as I see neoconf configure settings
so it is possible to change root_dir in neoconf?

**Reply by jhartum** [reply]
in vscode root_dir set in .python.env file, but how I can do the same in neoconf?

**Reply by folke** [reply]
You can still set it with lspconfig, or you can create a `.neoconf.json` file in your root dir

**Reply by jhartum** [reply]
I create .neoconf.json in my root dir and import work as well, but other stuff like mypy and ruff just stop working
it is possible to use settings from .vscode/settings.json and set root dir another directory?

**Reply by jhartum** [reply]
I already try set root_dir with lspconfig(you can see it on top of issue) but its not working
